components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5491,10 +5491,10 @@ private String getOpenAPIDefinitionForDeployment(Identifier apiId, String synaps
         String apiTenantDomain;
         String updatedDefinition = null;
         Map<String,String> hostsWithSchemes;
-        String definition = super.getOpenAPIDefinition(apiId);
+        String definition = super.getOpenAPIDefinition(apiId, null);
```

Comment:
shall we pass the proper tenant domain?

Reply:
https://github.com/wso2/carbon-apimgt/pull/9658/commits/142cd9cad67662fc64ac6e886127be649322fd43